docs/developing.rst

```diff
@@ -742,3 +742,246 @@ Note the call to :func:`toil.job.Job.encapsulate` creates the
 :class:`toil.job.Job.EncapsulatedJob`.
 
 
+Deploying a workflow
+====================
+
+If a Toil workflow is run on a single machine, there is nothing special you
+need to do. You change into the directory containing your user script and
+invoke like any Python script::
+
+   $ cd my_project
+   $ ls
+   userScript.py …
+   $ ./userScript.py …
+
+This assumes that your script is executable and contains a *shebang*, i.e. a
+line of the form
+
+::
+
+   #!/usr/bin/python
+
+or equivalent. The script can have dependencies, as long as those are installed
+on the machine, either globally, in a user-specific location or in a
+virtualenv. In the latter case, the virtualenv must of course be active when
+you run the user script.
+
+If, however, you want to run your workflow in a distributed environment, on
+multiple worker machines, either in the cloud or on a bare-metal cluster, your
+script needs to be made available to those other machines. If your script
+imports other modules, those modules also need to be made available on the
+workers. Toil can automatically do that for you, with a little help on your
+part. We call this feature ``hot-deployment`` of a workflow.
+
+
+Hot-deployment without dependencies
+-----------------------------------
+
+Your script has no additional dependencies, it imports only modules that are
+shipped with Python. Toil will take your script, store it in the job store and
+just before the jobs in your script are about to be run on a worker machine,
+your script will be saved to a temporary directory on the worker and loaded
+into the Python interpreter from there::
+
+   $ cd my_project
+   $ ls
+   userScript.py
+   $ ./userScript.py --batchSystem=mesos …
```

Comment:
You should note that this is the command line that the user is executing, not a script that Toil is invoking on the cluster. The context makes it read as if this is Toil running from the temporary directory on the worker. Also, I'd say something like:

> You may have noticed the addition of the `--batchSystem=...` flag...

It's obvious for power users, but prolly wouldn't be obvious to someone new to Toil.